shap/plots/_beeswarm.py

```diff
@@ -572,6 +573,9 @@ def summary_legacy(
         # if out_names is None: # TODO: waiting for slicer support of this
         #     out_names = shap_exp.output_names
 
+    if shap_exp is not None and len(shap_exp.base_values.shape) == 2 and shap_exp.base_values.shape[1] > 2:
+        shap_values = [shap_values[:, :, i] for i in range(shap_exp.base_values.shape[1])]
```

Comment:
Thank you for your suggestion. I have implemented the change.

Since this is my first time using the SHAP library, I'm unsure whether the `shap_exp` will be needed for further adjustments, as transitioning from an ndarray to an explanation object represents a significant change.

Nonetheless, it seems more logical to move this into the `if Explanation` block for now.